### PR TITLE
unix,fs: code cleanup of uv_fs_event_start for AIX

### PR DESCRIPTION
**Pull request by iWuzHere**
Removes unused variables and makes it easier to read and step through.

### CODE REVIEW

File: src/unix/aix.c [hunk 1]
```diff
@@ -782,53 +782,26 @@ int uv_fs_event_start(uv_fs_event_t* handle,
...
-      p++;
-    }
-    /* Copy to filename buffer */
-    while(filename[i] != NULL) {
-      *p = filename[i];
-      i++;
-      p++;
-    }
+    /* We have a relative pathname, compose the absolute pathname */
+    snprintf(cwd, PATH_MAX-1, "/proc/%lu/cwd", (unsigned long) getpid());
```
Comment (bnoordhuis) [inline src/unix/aix.c]: Can you use `sizeof(cwd) - 1`?

File: src/unix/aix.c [hunk 2]
```diff
@@ -782,53 +782,26 @@ int uv_fs_event_start(uv_fs_event_t* handle,
...
-    absolute_path[i] = 0;
-    fname[i] = 0;
-  }
-  i = 0;
 
   /* Figure out whether filename is absolute or not */
   if (filename[0] == '/') {
-    /* We have absolute pathname, create the relative pathname*/
+    /* We have absolute pathname */
     sprintf(absolute_path, filename);
```
Comment (bnoordhuis) [inline src/unix/aix.c]: Can you change this to `snprintf()` while you're here?

Comment (bnoordhuis) [inline src/unix/aix.c]: On second thought, using any of the `s*printf()` functions here is a format string vulnerability waiting to happen...

File: src/unix/aix.c [hunk 3]
```diff
@@ -782,53 +782,26 @@ int uv_fs_event_start(uv_fs_event_t* handle,
...
+    /* We have a relative pathname, compose the absolute pathname */
+    snprintf(cwd, PATH_MAX-1, "/proc/%lu/cwd", (unsigned long) getpid());
+    res = readlink(cwd, absolute_path, sizeof(absolute_path) - 1);
+    if (res < 0)
+      return res;
+
+    if (filename[0] == '.' && filename[1] == '/')
+      str_offset = 2;
+
+    strcat(absolute_path, filename + str_offset);
```
Comment (bnoordhuis) [inline src/unix/aix.c]: You need to zero-terminate `absolute_path` first, `readlink()` won't do it for you.

File: src/unix/aix.c (hunk 1 above)
Comment (iWuzHere) [inline src/unix/aix.c]: No you woudn't need to subtract one.

File: src/unix/aix.c (hunk 2 above)
Comment (iWuzHere) [inline src/unix/aix.c]: What about a ```strcpy```?

File: src/unix/aix.c (hunk 3 above)
Comment (iWuzHere) [inline src/unix/aix.c]: Does it matter? we are not using ```absolute_path``` right away and still need to append ```filename``` to it (which I assume is null terminated, I could be wrong). Though ```strcat`` does assume both strings are null terminated.

File: src/unix/aix.c (hunk 2 above)
Comment (bnoordhuis) [inline src/unix/aix.c]: Vulnerable to buffer overflows.  Let's go with `snprintf(absolute_path, sizeof(absolute_path), "%s", filename)`.

File: src/unix/aix.c (hunk 3 above)
Comment (bnoordhuis) [inline src/unix/aix.c]: Exactly.

Thinking about it more, `strcat()` has the same problem as `strcpy()`, it's susceptible to buffer overflows unless you carefully check there's enough room.  If you use a second buffer, you can `snprintf(result, sizeof(result), "%s%s", absolute_path, filename + str_offset)`.

**Comment by iWuzHere** [comment]
updated code as per comments

**Comment by iWuzHere** [comment]
Also realized there's no point having both ```rc``` and `res` since they do the same thing, so just removed `res`

File: src/unix/aix.c [hunk 4]
```diff
@@ -782,53 +782,29 @@ int uv_fs_event_start(uv_fs_event_t* handle,
...
+    rc = readlink(cwd, readlink_cwd, sizeof(readlink_cwd) - 1);
+    if (rc < 0)
+      return rc;
+    /* readlink does not null terminate our string */
+    readlink_cwd[rc] = '\0';
+
+    if (filename[0] == '.' && filename[1] == '/')
+      str_offset = 2;
+
+    snprintf(absolute_path, sizeof(absolute_path), "%s%s", readlink_cwd, filename + str_offset);
```
Comment (bnoordhuis) [inline src/unix/aix.c]: Long line.

**Comment by bnoordhuis** [comment]
LGTM with style nit.  CI: https://ci.nodejs.org/job/libuv-test-commit/31/

**Comment by iWuzHere** [comment]
Updated code.

From the CI:

```
[% 100|+ 277|-   4|T   0|S   5]: Done.
FAIL: test/run-tests
======================================================
```
https://ci.nodejs.org/job/libuv-test-commit-aix/nodes=aix61-ppc64/36/console

**Comment by bnoordhuis** [comment]
Thanks Imran, landed in 572d315.

**Pull request closed without merging**